engine/class_modules/sc_death_knight.cpp

```diff
@@ -2950,6 +2950,7 @@ struct death_knight_action_t : public Base
         if ( p() -> legendary.final_sentence.ok() )
         {
           p() -> buffs.final_sentence -> trigger();
+          p() -> replenish_rune( as<unsigned int>( p() -> find_spell( 353823 ) -> effectN( 1 ).resource( RESOURCE_RUNE ) ), p() -> gains.final_sentence );
```

Comment:
You should cache the rune amount to be replenished somewhere to avoid calling find_spell() during runtime